### PR TITLE
Release v0.4.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: v0.4.0

## Release details

Autonomy:
- Deprecates `autonomy analyse abci` command group has been deprecated
- Merges `autonomy analyse abci check-app-specs` and `autonomy analyse generate-app-specs` into `autonomy analyse fsm-specs`
- Moves `autonomy analyse abci check-handlers`  to `autonomy analyse handlers`  
- Moves `autonomy analyse abci docstrings` to `autonomy analyse docstrings`
- Moves `autonomy analyse abci logs` has been moved to `autonomy analyse logs`
- Refactors the FSM command defintion and extract the code to core and helper modules
- Updates the error messages on `autonomy build-image` command
- Improves error handling on `autonomy deploy` command group

Plugins:
- Renames the `tag` property to `image` on `aea_test_autonomy.docker.base.DockerImage` class

Packages:
- Updates the abstract round abci skill to ignore tendermint blocks with a height lower than the initial height of the tendermint chain
- Adds add yield statement in `CheckTransactionHistoryBehaviour._check_tx_history` to avoid freezing the entire behaviour
- Moves the tendermint healthcheck from individual round behaviours to the `AbstractRoundBehaviour` to ensures that tendermint would reset in cases when the communication is unhealthy

Tests:
- Increases the sleep time in `test_async_behaviour_sleep` and `TestRegistrationStartupBehaviour` tests to avoid flakiness

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] I've updated the dependencies versions to the latest, wherever is possible.
- [ ] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [ ] I built the documentation and updated it with the latest changes
- [ ] I've added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+- Updates the error messages on `autonomy build-image` command
+- Improves error handling on `autonomy deploy` command group
+
+Plugins:
+- Renames the `tag` property to `image` on `aea_test_autonomy.docker.base.DockerImage` class
+
+Packages:
+- Updates the abstract round abci skill to ignore tendermint blocks with a height lower than the initial height of the tendermint chain
+- Adds add yield statement in `CheckTransactionHistoryBehaviour._check_tx_history` to avoid freezing the entire behaviour
+- Moves the tendermint healthcheck from individual round behaviours to the `AbstractRoundBehaviour` to ensures that tendermint would reset in cases when the communication is unhealthy
```
[the remark this answers is not in the document]
Comment (Adamantios) [reply HISTORY.md:25]: Update, I see you added these notes already in the plugins!

File: HISTORY.md [hunk 2]
```diff
@@ -1,5 +1,34 @@
...
+
+Tests:
+- Increases the sleep time in `test_async_behaviour_sleep` and `TestRegistrationStartupBehaviour` tests to avoid flakiness
+- Tests for newly introduced code in the `ApiSpecs` implementation
+
```
Comment (Karrenbelt) [inline HISTORY.md:30]: tests for `abstract_round_abci/test_tools/integration` as well - https://github.com/valory-xyz/open-autonomy/pull/1561
which includes a breaking change (renaming `_HarHatHelperIntegration` to `HardHatHelperIntegration`) https://github.com/valory-xyz/open-autonomy/pull/1561/files#r1024249883

File: HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,34 @@
 # Release History - `open-autonomy`
 
+
+# 0.4.0 (2022-11-17)
+
+Autonomy:
+- Deprecates `autonomy analyse abci` command group has been deprecated
```
Comment (DavidMinarsch) [inline HISTORY.md:7]:
```suggestion
- Deprecates `autonomy analyse abci` command group
```

File: HISTORY.md [hunk 4]
```diff
@@ -1,5 +1,34 @@
...
+- Improves error handling on `autonomy deploy` command group
+
+Plugins:
+- Renames the `tag` property to `image` on `aea_test_autonomy.docker.base.DockerImage` class
+- Extends `ApiSpecs` to support getting the response from a list
+- Adds better error `logging` in the `ApiSpecs`
+- Adds backoff logic to the `ApiSpecs`
+
+Packages:
+- Updates the abstract round abci skill to ignore tendermint blocks with a height lower than the initial height of the tendermint chain
```
Comment (DavidMinarsch) [inline HISTORY.md:23]:
```suggestion
- Updates the `abstract_round_abci` skill to ignore Tendermint blocks with a height lower than the initial height of the Tendermint chain
```

File: tests/test_docs/test_commands.py [hunk 5]
```diff
@@ -158,6 +158,8 @@ def test_validate_doc_commands() -> None:
 
     # Validate all matches
     for file_ in target_files:
+        if file_.name == "upgrading.md":
+            continue
```
Comment (Karrenbelt) [inline tests/test_docs/test_commands.py:162]: This is excluded now because deprecated commands (and their replacement) are listed I suppose.

**Review by Adamantios — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**